stdlib/Random/src/misc.jl

```diff
@@ -220,6 +220,20 @@ function shuffle!(r::AbstractRNG, a::AbstractArray)
     return a
 end
 
+function shuffle!(r::AbstractRNG, a::AbstractArray{Bool})
+    old_count = count(a)
+    x = 2old_count <= length(a)
+    fuel = x ? old_count : length(a) - old_count
+    fuel == 0 && return a
+    a .= !x
+    while 0 < fuel
```

Comment:
```suggestion
    while fuel > 0
```
nitpick, but it seems easier to read to me